### PR TITLE
Add Makefile and unit tests

### PR DESCRIPTION
**Pull request by smaye81**
This adds a Makefile and unit tests to every project so that we can quickly run tests, builds, and library updates with a single make command.

### CODE REVIEW

File: react/parcel/package.json [hunk 1]
```diff
@@ -4,22 +4,27 @@
   "description": "",
   "source": "src/index.html",
   "scripts": {
-    "start": "npm run buf:generate && npx parcel src/index.html -p 3000",
-    "build": "npm run buf:generate && npx parcel build src/index.html",
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "unset NODE_OPTIONS && npm run buf:generate && npx parcel src/index.html -p 3000 --log-level error",
```
Comment (timostamm) [inline react/parcel/package.json:7]: Why do we `unset NODE_OPTIONS`?

File: angular/package.json [hunk 2]
```diff
@@ -2,12 +2,12 @@
   "name": "buf-ng",
   "version": "0.0.0",
   "scripts": {
-    "start": "ng serve --port 3000",
-    "build": "ng build",
-    "watch": "ng build --watch --configuration development",
-    "test": "ng test",
+    "start": "npx ng serve --port 3000",
```
Comment (timostamm) [inline angular/package.json:5]: I'm surprised to see `npx` in an npm script. npm script already inherit from the npm environment, e.g. have "binaries" on the $PATH. Is there a specific reason we're adding this?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.tmp
+            .tmp
+          key: ${{ runner.os }}-connect-web-integration-ci-${{ hashFiles('Makefile') }}
+          restore-keys: |
+            ${{ runner.os }}-connect-web-integration-ci-
+      - name: make
+        run: make
```
Comment (timostamm) [inline .github/workflows/ci.yaml:24]: Would it be a bit more clear that this runs only tests (and not the updates) if it was `make test`?

**Review by timostamm — COMMENTED** [review]
This will be incredibly helpful in maintaining this repository 🙏

File: react/parcel/package.json (hunk 1 above)
Comment (smaye81) [reply react/parcel/package.json:7]: React Native scripts require it to be set like `export NODE_OPTIONS=--openssl-legacy-provider`, but if it's still set for some of the other projects, the command errors out, so it needs unset.  If there's a better way to do this, I'm all ears.  We could do it in the Makefile, but then it leaves the package.json files vulnerable to this error.

File: angular/package.json (hunk 2 above)
Comment (smaye81) [reply angular/package.json]: Ah no reason.  I removed the `npx` occurrences from all the package.jsons.

File: react/parcel/package.json (hunk 1 above)
Comment (timostamm) [reply react/parcel/package.json]: Can we somehow only set NODE_OPTIONS for react-native then? I think users will wonder about `unset NODE_OPTIONS` when they look up this parcel example or others.

Comment (smaye81) [reply react/parcel/package.json]: Yep, I removed the unsetting in the scripts using it.  The issue could still occur if you manually set it and then run parcel or vue, but the overall `make test` run seems to work.

Comment (timostamm) [reply react/parcel/package.json]: I think most users will come here for a specific framework. Seems fine to me.

**Review by timostamm — APPROVED** [review]

**Pull request merged**